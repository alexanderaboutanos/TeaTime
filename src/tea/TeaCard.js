/** @format */

import { Col, Card, Button } from "react-bootstrap";

const TeaCard = ({
  id,
  title,
  brand,
  img_url,
  description,
  origin,
  btn1,
  btn2,
}) => {
  let btn1Txt;
  let btn2Txt;
  if (origin === "DiscoverTeaList") {
    btn1Txt = "+ wish list";
    btn2Txt = "+ my teas";
  }
  if (origin === "MyTeaList") {
    btn1Txt = "+ wish list";
    btn2Txt = "delete tea";
  }
  if (origin === "WishList") {
    btn1Txt = "+ my teas";
    btn2Txt = "delete tea";
  }

  return (
    <Col>
      <Card id={id}>
        <Card.Img variant="top" src={img_url} />
        <Card.Body>
          <Card.Title>{title}</Card.Title>
          <Card.Text>{brand}</Card.Text>
          <Card.Text>{description}</Card.Text>
        </Card.Body>
        <Card.Body>
          <>
            <Button onClick={() => btn1(id)} size="sm" variant="primary">
              {btn1Txt}
            </Button>
            <Button onClick={() => btn2(id)} size="sm" variant="primary">
              {btn2Txt}
            </Button>
          </>
        </Card.Body>
      </Card>
    </Col>
  );
};

export default TeaCard;
