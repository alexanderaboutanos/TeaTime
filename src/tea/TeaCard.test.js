/** @format */

import React from "react";
import { render } from "@testing-library/react";
import { MemoryRouter } from "react-router";
import TeaCard from "./TeaCard";

it("renders without crashing", function () {
  render(
    <MemoryRouter>
      <TeaCard />
    </MemoryRouter>
  );
});

it("matches snapshot", function () {
  const { asFragment } = render(
    <MemoryRouter>
      <TeaCard />
    </MemoryRouter>
  );
  expect(asFragment()).toMatchSnapshot();
});
