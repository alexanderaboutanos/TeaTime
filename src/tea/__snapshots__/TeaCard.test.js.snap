// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="col"
  >
    <div
      class="h-100 card"
    >
      <img
        class="card-img-top"
        style="cursor: pointer;"
      />
      <div
        class="bg-light card-body"
        style="cursor: pointer;"
      >
        <div
          class="text-truncate card-title h5"
        />
        <p
          class="card-text"
        />
        <p
          class="crop-text-2 card-text"
        />
      </div>
      <div
        class="card-footer"
      >
        <button
          class="btn btn-primary btn-sm"
          style="margin: 3px;"
          type="button"
        />
        <button
          class="btn btn-primary btn-sm"
          type="button"
        />
      </div>
    </div>
  </div>
</DocumentFragment>
`;
